### ISSUE TITLE
"Expected *.h to be first header included." errors in custom 4.15

### ISSUE THREAD

**Issue by eksod** [20tab/UnrealEnginePython#214]
Hello!

I was trying to add UnrealEnginePython to our custom version of 4.15, but I get a bunch of errors like these when compiling the project and engine:

`1>c:\Code\UE4\4.15-dw\Engine\Plugins\UnrealEnginePython\Source\UnrealEnginePython\Private\PyCharacter.cpp(1): error : Expected PyCharacter.h to be first header included.
1>c:\Code\UE4\4.15-dw\Engine\Plugins\UnrealEnginePython\Source\UnrealEnginePython\Private\PyCommandlet.cpp(1): error : Expected PyCommandlet.h to be first header included.
1>c:\Code\UE4\4.15-dw\Engine\Plugins\UnrealEnginePython\Source\UnrealEnginePython\Private\PyHUD.cpp(1): error : Expected PyHUD.h to be first header included.`

If I simply move `#include "PyCharacter.h"` to the top of the include list that file gets compiled, but I was wondering if I should go with that or not. I guess this is related to [IWYU changes](https://docs.unrealengine.com/latest/INT/Programming/UnrealBuildSystem/IWYUReferenceGuide/)?

Thanks!

**Comment by rdeioris** [comment]
Hi, if you have enabled IWYU it could happen. To be compatible with older version we need to have the PCH header on top. The strange thing is that you should get that error for basically every UObject class of the project, not only for those 3. Maybe the HeaderTool complains because more includes are called in that files ?

**Comment by eksod** [comment]
Yes yes, I do get for ALL the files in the plugin, I just pasted the first three for readability here. :)

What would you recommend to do on this case? I could change locally, but it would make it a bit more painful when we would update the files...

**Comment by rdeioris** [comment]
I will check if i can add some #ifdef or eventually make the older Header tools happy even when not defining the PCH file

**Comment by eksod** [comment]
Great! Grazie mille!

**Comment by UnNeill** [comment]
I get this with 4.18.2.  the other source plugins and engine build fine.
is there any easy fix?
thanks

**Comment by UnNeill** [comment]
@rdeioris i'm trying to compile the unrealpython plugin for 4.18.2 and I get the same error.  I changed the order of all of the header includes but then had other problems.  Ive built engine source and the houdini engine plugin with vs2017 community (MSBuild 15.0) while the UnrealPython tools/build_win64.py script references MSBuild 14.0.  Does this build with vs2017?  
I edited the build script to use MSBuild 15 but i got the same errors as i did in the IDE.
What I am doing wrong?
